### ISSUE TITLE
Error in system/setup

### ISSUE THREAD

**Issue by spencergibb** [ansible/ansible#6382]
##### Issue Type:

Bug Report
##### Ansible Version:

1.5.1
##### Environment:

What OS are you running Ansible from and what OS are you managing?  from Mac OSX 10.8 managing Mac OSX 10.8
##### Summary:

system/setup fails with a python error
##### Steps To Reproduce:

ran `ansible-playbook -i hosts playbook.yml`

playbook.yml

```

---
- hosts: workstation

  tasks:
     - name: print from playbook
        debug: msg="in playbook.yml"
```

hosts

```
[workstation]
localhost               ansible_connection=local
```
##### Expected Results:

successful running of a playbook with just a debug task
##### Actual Results:

```
failed: [localhost] => {"failed": true, "parsed": false}
invalid output was: Traceback (most recent call last):
  File "/Users/spencergibb/.ansible/tmp/ansible-tmp-1394499808.68-243093264038607/setup", line 3443, in     <module>
    main()
  File "/Users/spencergibb/.ansible/tmp/ansible-tmp-1394499808.68-243093264038607/setup", line 2360, in main
    data = run_setup(module)
  File "/Users/spencergibb/.ansible/tmp/ansible-tmp-1394499808.68-243093264038607/setup", line 2301, in run_setup
    facts = ansible_facts(module)
  File "/Users/spencergibb/.ansible/tmp/ansible-tmp-1394499808.68-243093264038607/setup", line 2291, in ansible_facts
    facts.update(Network(module).populate())
TypeError: __init__() takes exactly 1 argument (2 given)
```

**Comment by fesplugas** [comment]
Same here, version `1.5.2` is also affected.

https://github.com/ansible/ansible/commit/ba0fec4f#commitcomment-5639260

**Comment by careyrp** [comment]
The problem is the code for class LinuxNetwork was modified to take parameter module, but all other sub-classes of class Network were not modified. So the code "Network(module)" on all platforms except platforms that use LinuxNetwork will fail.

Note:
The Network class contains a **new** function. This will pick the appropriate sub-class of Network and instantiates the sub-class instead of instantiating the Network class directly. So the code "Network(module)" will create an object of class LinuxNetwork.

Solution:
Modify the **init** method for the following classes to accept a module parameter:
GenericBsdIfconfigNetwork
DarwinNetwork
FreeBSDNetwork
AIXNetwork
OpenBSDNetwork
SunOSNetwork

def **init**(self, module):
        self.module = module

**Comment by mpdehaan** [comment]
Looking at this now, thanks

**Comment by careyrp** [comment]
This issue should be closed. Fixed in release 1.5.3.

**Comment by jctanner** [comment]
@careyrp thanks!
